pkg/resourcemapping/operator_resource_mapping.go

```diff
@@ -0,0 +1,340 @@
+			if !ok {
+				glog.Errorf("resourceMappingTemplate does not have the expected 'map[string]interface{}' structure in ORM CR %s", ormCRName)
+				continue
+			}
+			rmTemplates = append(rmTemplates, rmTemplate)
+		}
+		for _, componentName := range componentNames {
+			componentKey := srcResourceKind + "/" + componentName
+			ormSpec := ORMTemplate{
+				componentName:            componentName,
+				resourceMappingTemplates: rmTemplates,
+			}
+			ormTemplateMap[componentKey] = ormSpec
+		}
+	}
+	return ormTemplateMap, nil
+}
+
+// Update updates the corresponding CR for an Operator manged resource based on OperatorResourceMapping
+// origControllerObj -- original K8s controller object
+// updatedControllerObj -- updated K8s controller object based on Turbo actionItem, from which the resource value is fetched
+//                         and will be set to the corresponding CR
+// controllerOwnerReference -- ownerReference of a K8s controller, which contains metadata of a Operator CR
+func (ormClient *ORMClient) Update(origControllerObj, updatedControllerObj *unstructured.Unstructured, controllerOwnerReference metav1.OwnerReference) error {
+	operatorCRUID := string(controllerOwnerReference.UID)
+	componentKey := updatedControllerObj.GetKind() + "/" + updatedControllerObj.GetName()
+	ormClient.cacheLock.Lock()
+	defer ormClient.cacheLock.Unlock()
+	operatorResourceSpec, exists := ormClient.operatorResourceSpecMap[operatorCRUID]
+	if !exists {
+		return fmt.Errorf("operatorResourceSpec not found in operatorResourceSpecMap for operatorCR %s", operatorCRUID)
+	}
+	ormTemplate, exists := operatorResourceSpec.ormTemplateMap[componentKey]
+	if !exists {
+		return fmt.Errorf("ormTemplate not found in ormTemplateMap for componentKey %s for operatorCR %s", componentKey, operatorCRUID)
+	}
+
+	operatorResKind := controllerOwnerReference.Kind
+	operatorResName := controllerOwnerReference.Name
+	operatorRes := operatorResKind + "/" + operatorResName
+
+	resourceNamespace := updatedControllerObj.GetNamespace()
+	dynResourceClient := ormClient.dynClient.Resource(operatorResourceSpec.operatorGVResource).Namespace(resourceNamespace)
+	operatorCR, err := dynResourceClient.Get(operatorResName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to get CR %s for %s in namespace %s: %v", operatorRes, componentKey, resourceNamespace, err)
+	}
+
+	glog.V(2).Infof("Updating CR %s for %s in namespace %s", operatorRes, componentKey, resourceNamespace)
+	updated := false
+	// Update based on each resourceMappingTemplate
+	for _, resourceMappingTemplate := range ormTemplate.resourceMappingTemplates {
+		// Set resourceMappingComponentName to resourceMappingTemplate so as to parse the srcPath and destPath based on text template
+		resourceMappingTemplate[resourceMappingComponentName] = ormTemplate.componentName
+		srcPath, destPath, err := ormClient.parseSrcAndDestPath(resourceMappingTemplate)
```

Comment:
Would this work if the component name is skipped and the `src` and `dst` paths are actual paths. For example to control an operator which has only 1 component. Or alternatively for another CR writer (user) who specifies the paths in which nothing needs to be replaced.

Reply:
Good point. Yes, this will still work if nothing needs to be replaced in the `srcPath` and `destPath`, for example:
```
.spec.template.spec.container.0.resources
```
or
```
.spec.resources
```
I'll add a new unit test for this case

Reply:
Awsome!